### PR TITLE
Feature/ethernet rtc alt

### PR DESCRIPTION
**Pull request by jr1221**
## Changes

_Explanation of changes goes here_

Uses the ethernet clock rather than the RTC.  Derived from pll1_q for better accuracy.

## Notes

_Any other notes go here_

## Test Cases

- Case A
- Edge case
- ...

## To Do

_Any remaining things that need to get done_

- [ ] item 1
- [ ] ...

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please reach out to your Project Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [ ] No merge conflicts
- [ ] All checks passing
- [ ] Remove any non-applicable sections of this template
- [ ] Assign the PR to yourself
- [ ] Request reviewers & ping on Slack
- [ ] PR is linked to the ticket (fill in the closes line below)

Closes # (issue #)

### CODE REVIEW

**Comment by jr1221** [comment]
Test cases: Time eyeball looks accurate to system clock.  Calibration can occur once we have access to the eth_pps pin.

File: NetX/src/u_rtc.c [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+			// clk_ptp_ref_i comes from pll1_q_clk (eth clock) (pg 456)
+			// this cool API gives us the frequency (in hz)
+
+			// for some reason we need to enable pll1q output?
+			SET_BIT(RCC->PLL1CFGR, RCC_PLL1CFGR_PLL1QEN);
+
+			PLL1_ClocksTypeDef pll1;
+			HAL_RCCEx_GetPLL1ClockFreq(&pll1);
+			if (1000000000 % pll1.PLL1_Q_Frequency != 0) {
+				// Should this be an assertion instead?
```
Comment (caiodasilva2005) [inline NetX/src/u_rtc.c]: keep no assertion here, warning should be fine

File: NetX/src/u_rtc.c [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+			if (time_ptr->nanosecond >= 0) {
+				// this will automatically clear the ADDSUB so we add
+				WRITE_REG(heth.Instance->MACSTNUR,
+					  time_ptr->nanosecond);
+			} else {
+				// take complement starting from one million
+				uint32_t complement =
+					1000000000 - abs(time_ptr->nanosecond);
+				WRITE_REG(heth.Instance->MACSTNUR, complement);
+				SET_BIT(heth.Instance->MACSTNUR,
```
Comment (caiodasilva2005) [inline NetX/src/u_rtc.c:137]: maybe add note that this will sub from update reg

File: NetX/src/u_rtc.c [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+			time_ptr->nanosecond = READ_REG(heth.Instance->MACSTNR);
+			time_ptr->second_low = READ_REG(heth.Instance->MACSTSR);
+			time_ptr->second_high = 0;
+			TX_RESTORE
+			break;
+		case NX_PTP_CLIENT_CLOCK_ADJUST:
+			/// A small adjustment in time (<1 second)
+			TX_DISABLE
+			// use add, not overwrite.
+			// MACTSUR must be 0 at this point, not wasting CPU time though
```
Comment (caiodasilva2005) [inline NetX/src/u_rtc.c:126]: can we print a warning if this is called with MACTSUR as 0

Comment (jr1221) [reply NetX/src/u_rtc.c:126]: good question.  I would prefer not (else I would just clear it every time) because literally every CPU cycle is one more nanosecond we are delayed, even just the if statement I would get concerned.
